File: api/health.ts
import { NextRequest } from 'next/server';
import { createClient } from '@libsql/client';

export const config = { runtime: 'edge' };

export default async function handler(_req: NextRequest) {
  try {
    const url = process.env.TURSO_DATABASE_URL;
    const authToken = process.env.TURSO_AUTH_TOKEN;
    if (url && authToken) {
      const client = createClient({ url, authToken });
      // simple ping
      await client.execute('SELECT 1');
    }
    return new Response(JSON.stringify({ success: true, data: 'OK', error: null }), {
      status: 200,
      headers: { 'content-type': 'application/json' }
    });
  } catch (e: any) {
    return new Response(JSON.stringify({ success: false, data: null, error: e?.message || 'error' }), {
      status: 500,
      headers: { 'content-type': 'application/json' }
    });
  }
}

File: api/health.rs
use vercel_runtime::{run, Body, Error, Request, Response, StatusCode};
use serde::Serialize;

#[derive(Serialize)]
struct ApiResponse<'a, T: Serialize> {
    success: bool,
    data: Option<T>,
    error: Option<&'a str>,
}

#[tokio::main]
async fn main() -> Result<(), Error> {
    run(handler).await
}

async fn handler(_req: Request) -> Result<Response<Body>, Error> {
    let payload = ApiResponse { success: true, data: Some("OK"), error: None };
    let body = serde_json::to_vec(&payload).unwrap();
    Ok(Response::builder()
        .status(StatusCode::OK)
        .header("content-type", "application/json")
        .body(Body::Binary(body))
        .unwrap())
}